pkg/server/db_options.go

```diff
@@ -59,12 +59,26 @@ type dbOptions struct {
 
 	IndexOptions *indexOptions `json:"indexOptions"`
 
+	AutoOpen featureState `json:"autoOpen"` // unspecfied is considered as enabled for backward compatibility
+
 	CreatedBy string    `json:"createdBy"`
 	CreatedAt time.Time `json:"createdAt"`
 	UpdatedBy string    `json:"updatedBy"`
 	UpdatedAt time.Time `json:"updatedAt"`
 }
 
+type featureState int
+
+const (
+	unspecifiedState featureState = iota
+	enabledState
+	disabledState
```

Comment:
Since this i persisted I think it's better to set explicit values here to avoid conincidental reordering / modifications of those values:

```suggestion
	unspecifiedState featureState = 0
	enabledState featureState = 1
	disabledState featureState = 2
```